server/src/main/java/io/crate/geo/GeoJSONUtils.java

```diff
@@ -155,6 +181,81 @@ public static Object map2LuceneShape(Map<String, Object> geoJSONMap) {
         return geoJSONString2ShapeBuilder(geoJSONMap).buildLucene();
     }
 
+    /**
+     * Checks equality, returns 0 for equal shapes and 1 for non-equal.
+     * This method doesn't do any validation and assumes that Maps represent correct geo shapes.
+     * When comparing geometries collections, order of geometries is important,
+     * ie collection(point1, point2) != collection(point2, point1).
+     */
+    @SuppressWarnings("unchecked")
+    public static int compare(Map<String, Object> val1, Map<String, Object> val2) {
+        Object type1 = val1.get(TYPE_FIELD);
+        Object type2 = val1.get(TYPE_FIELD);
+        assert type1 != null && type2 != null : "Shapes must be valid"; // implicitCast and sanitizeValue validate provided values.
+        if (type1.equals(type2) == false) {
+            return 1;
+        }
+
+        int result;
+        if (GEOMETRY_COLLECTION.equals(type1)) {
+            // Both shapes are valid geometries collections.
+            Object geometries1 = val1.get(GeoJSONUtils.GEOMETRIES_FIELD);
+            Object geometries2 = val2.get(GeoJSONUtils.GEOMETRIES_FIELD);
+            Iterator<?> it1 = iterator(geometries1);
+            Iterator<?> it2 = iterator(geometries2);
```

Comment:
Afaik geometries can't be an array but must be a collection, so could use assert/cast here instead of going through `iterator`